src/base/components/DeviceProfile.js

```diff
@@ -2,11 +2,146 @@ import React from 'react';
 class DeviveProfileTab extends React.Component {
     constructor(props) {
         super(props);
+        this.state = {         
+          devicestatus: "off",
+          temperature: "0",
+          testbed_network: "0",
+          tx_level: "0",
+          cha: "0",
+          tx_mode:"0",
+          rx_mode:"0",
+          rssi: "0",
+          lqi: "0",
+          nodeid: "0"          
+        }
+    }
+    componentDidMount() {
+      this.getSensorData();      
     }
 
+    setConfig = () => {
+      var setval={dev:"Mote",opr:this.state.devicestatus};    
+      fetch("https://immense-temple-14028.herokuapp.com/api/Device", {
+        method: "POST",
+        dataType: "JSON",
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+          'Content-Type': 'application/json',
+          'Accept': 'application/json'
+        },
+        body: JSON.stringify(setval)
+      })
+      .then((resp) => {
+        return resp.json()
+      })
+      .then((data) => {
+        if(this.state.devicestatus === "on")
+          this.setState({ devicestatus: "off"});
+        else
+        this.setState({ devicestatus: "on"});
+      })
+      .catch((error) => {
+        console.log(error, "catch the hoop")
+      })
+    }
+    setTxConfig = (event) => {      
+      this.setState({tx_level: event.target.value});
+      var setval={dev:"Mote",opr:"SETTX="+event.target.value+"\n"};
+      
+      fetch("https://immense-temple-14028.herokuapp.com/api/Device", {
+        method: "POST",
+        dataType: "JSON",
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+          'Content-Type': 'application/json',
+          'Accept': 'application/json'
+        },
+        body: JSON.stringify(setval)
+      })
+      .then((resp) => {
+        return resp.json()
+      })
+      .then((data) => {
+        //this.setTxVal(dataval.txLevel);        
+      })
+      .catch((error) => {
+        console.log(error, "catch the hoop")
+      })
+    }
+    setChConfig = (event) => {      
+      var setval={dev:"Mote",opr:"SETCH="+event.target.value+"\n"};
+      this.setState({cha: event.target.value});      
+      fetch("https://immense-temple-14028.herokuapp.com/api/Device", {
+        method: "POST",
+        dataType: "JSON",
+        headers: {
+          "Access-Control-Allow-Origin": "*",
+          'Content-Type': 'application/json',
+          'Accept': 'application/json'
+        },
+        body: JSON.stringify(setval)
+      })
+      .then((resp) => {
+        return resp.json()
+      })
+      .then((data) => {
+        console.log("here data is ",data);
+      })
+      .catch((error) => {
+        console.log(error, "catch the hoop")
+      })
+    }
+    
+    getSensorData = () => {
+      var setval={dev:"Mote",opr:this.state.devicestatus};
+      //console.log("Here object is "+JSON.stringify(object));
+      fetch("https://immense-temple-14028.herokuapp.com/api/Device/GetSensorValue", {
```

Comment:
The same

Reply:
url will be taken from config file in future....